compiler/testSuite/FVfmod00.bal

```diff
@@ -0,0 +1,148 @@
+    io:println(isNaN(-0.0 % 0.0)); // @output true
+    io:println(isNaN(nInf % 0.0)); // @output true
+    io:println(isNaN(-nan % 0.0)); // @output true
+
+    io:println(isEqual(7.0 % pInf, 7.0)); // @output true
+    io:println(isEqual(5.0 % pInf, 5.0)); // @output true
+    io:println(isEqual(0.7 % pInf, 0.7)); // @output true
+    io:println(isEqual(0.5 % pInf, 0.5)); // @output true
+    io:println(isEqual(0.0 % pInf, 0.0)); // @output true
+    io:println(isEqual(-7.0 % pInf ,-7.0)); // @output true
+    io:println(isEqual(-5.0 % pInf ,-5.0)); // @output true
+    io:println(isEqual(-0.7 % pInf ,-0.7)); // @output true
+    io:println(isEqual(-0.5 % pInf ,-0.5)); // @output true
+    io:println(isEqual(-0.0 % pInf ,-0.0)); // @output true
+
+    io:println(isEqual(7.0 % nInf, 7.0)); // @output true
+    io:println(isEqual(5.0 % nInf, 5.0)); // @output true
+    io:println(isEqual(0.7 % nInf, 0.7)); // @output true
+    io:println(isEqual(0.5 % nInf, 0.5)); // @output true
+    io:println(isEqual(0.0 % nInf, 0.0)); // @output true
+    io:println(isEqual(-7.0 % nInf ,-7.0)); // @output true
+    io:println(isEqual(-5.0 % nInf ,-5.0)); // @output true
+    io:println(isEqual(-0.7 % nInf ,-0.7)); // @output true
+    io:println(isEqual(-0.5 % nInf ,-0.5)); // @output true
+    io:println(isEqual(-0.0 % nInf ,-0.0)); // @output true
+
+    io:println(isEqual(7.0 % 7.0, 0.0)); // @output true
+    io:println(isEqual(5.0 % 7.0, 5.0)); // @output true
+    io:println(isEqual(0.7 % 7.0, 0.7)); // @output true
+    io:println(isEqual(0.5 % 7.0, 0.5)); // @output true
+    io:println(isEqual(0.0 % 7.0, 0.0)); // @output true
+    io:println(isEqual(-7.0 % 7.0 ,-0.0)); // @output true
+    io:println(isEqual(-5.0 % 7.0 ,-5.0)); // @output true
+    io:println(isEqual(-0.7 % 7.0 ,-0.7)); // @output true
+    io:println(isEqual(-0.5 % 7.0 ,-0.5)); // @output true
+    io:println(isEqual(-0.0 % 7.0 ,-0.0)); // @output true
+
+    io:println(isEqual(7.0 % 5.0, 2.0)); // @output true
+    io:println(isEqual(-7.0 % 5.0 ,-2.0)); // @output true
+
+    io:println(isEqual(7.0 % -7.0, 0.0)); // @output true
+    io:println(isEqual(5.0 % -7.0, 5.0)); // @output true
+    io:println(isEqual(0.7 % -7.0, 0.7)); // @output true
+    io:println(isEqual(0.5 % -7.0, 0.5)); // @output true
+    io:println(isEqual(0.0 % -7.0, 0.0)); // @output true
+    io:println(isEqual(-7.0 % -7.0 ,-0.0)); // @output true
+    io:println(isEqual(-5.0 % -7.0 ,-5.0)); // @output true
+    io:println(isEqual(-0.7 % -7.0 ,-0.7)); // @output true
+    io:println(isEqual(-0.5 % -7.0 ,-0.5)); // @output true
+    io:println(isEqual(-0.0 % -7.0 ,-0.0)); // @output true
+
+    io:println(isEqual(7.0 % -5.0, 2.0)); // @output true
+    io:println(isEqual(-7.0 % -5.0 ,-2.0)); // @output true
+}
+
+function isNaN(float val) returns string {
+	if float:isNaN(val) {
+		return "true";
+	}
+	return "false";
+}
+
+function isEqual(float lhs, float rhs) returns string {
+    if lhs == rhs {
```

Comment:
should we use `===` here to catch `-0.0` cases ?

Reply:
Yes.

Reply:
I'll fix when I am moving this form `F` to `V`.